From 27512a789981d73158202de6a390762b2e71e4aa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ale=C5=A1=20Mat=C4=9Bj?= <amatej@redhat.com>
Date: Mon, 9 Jun 2025 09:25:39 +0200
Subject: [PATCH] Propagate return value from `prepare_repo_download_targets`

It also ensures that when return value is false there is some error set.
---
 librepo/yum.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/librepo/yum.c b/librepo/yum.c
index 56bca3e..482d4d9 100644
--- a/librepo/yum.c
+++ b/librepo/yum.c
@@ -961,7 +961,12 @@ lr_yum_download_repo(LrHandle *handle,
 
     assert(!err || *err == NULL);
 
-    prepare_repo_download_targets(handle, repo, repomd, NULL, &targets, &cbdata_list, err);
+    ret = prepare_repo_download_targets(handle, repo, repomd, NULL, &targets, &cbdata_list, err);
+    if (!ret) {
+        assert(!err || *err != NULL);
+        return ret;
+    }
+    assert(!err || *err == NULL);
 
     if (!targets)
         return TRUE;
-- 
2.52.0

